activator/activator.go

```diff
@@ -0,0 +1,90 @@
+package activator
+
+import (
+	"os"
+
+	guuid "github.com/google/uuid"
+
+	"v9_deployment_manager/database"
+	"v9_deployment_manager/log"
+	"v9_deployment_manager/worker"
+)
+
+type Activator struct {
+	driver *database.Driver
+}
+
+func CreateActivator(driver *database.Driver) (*Activator, error) {
+	return &Activator{
+		driver: driver,
+	}, nil
+}
+
+func (a *Activator) Activate(compID *worker.ComponentID, worker *worker.V9Worker) error {
+	// Get random tar name
+	tarName := guuid.New().String()
+	fullRepoName := compID.User + "/" + compID.Repo
+	// Get Repo Contents
+	log.Info.Println("Cloning Repo...")
+	clonedPath, err := cloneRepo(fullRepoName)
+	if err != nil {
+		log.Error.Println("Error cloning repo:", err)
+		return err
+	}
+	defer os.RemoveAll(clonedPath) // clean up
+
+	//FIXME git checkout HEAD/hash
+	if compID.Hash == "HEAD" {
+		compID.Hash, err = getHash(clonedPath)
+		if err != nil {
+			log.Error.Println("Error getting hash from repo:", err)
+			return err
+		}
+	}
+	// Setup the DB deploying entry
+	err = a.driver.EnterDeploymentEntry(compID)
```

Comment:
In future will want to customize the reason as a parameter

Reply:
like the reason for deployment? push/repo_added/install